server-x/README.md

```diff
@@ -0,0 +1,39 @@
+# Keycloak.X Docker Image
+
+The `Dockerfile` for Keycloak.X Docker Image.
+
+See the `Build` section for more details on how to build the image.
+
+Once built, you can run the server in the same manner as when using the Keycloak.X distribution by passing any command-line argument:
+
+    docker run --name keycloak -p 8080:8080 \
+           -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=change_me \
+           IMAGE[:TAG] \
+           start-dev
+
+## Auto-Configuration
+
+Given the immutability of containers, you can use the `--auto-config` option in order to apply any configuration when running a new container. Once the container is created, subsequent restarts will never go through the configuration phase again but just start the server with the configuration previously defined.
+
+    docker run --name keycloak -p 8080:8080 \
+           -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=change_me \
+           IMAGE[:TAG] \
+           --auto-config --db=postgres -Dkc.db.url.host=<DB_HOST> --db-username=keycloak --db-password=change_me --http-enabled=true
+
+The command above should be enough to run a server using a PostgreSQL database listening on a given `DB_HOST`.
+
+The recommendation is to always create your own image based on this image. By doing that, you are able to perform more customizations,
+such as deploying custom providers or themes, as well as improve the startup time by eliminating the configuration step.
+
+However, the `--auto-config` option should help when trying out Keycloak and for development purposes.
+
+## Build
+
+It is possible to download the Keycloak distribution from a URL:
+
+    docker build --build-arg KEYCLOAK_DIST=http://<HOST>:<PORT>/keycloak.x-<VERSION>.tar.gz -t <TAG> .
+
+For Keycloak built locally you need to first build the distribution then serve it with a web browser. For example use Python HTTP Server:
+
+    cd $KEYCLOAK_SOURCE/distribution/server-dist/target
```

Comment:
Fixed.

Configure the server after the container is created is harder. Because then you need to run commands within the container.

Also, containers are usually ephemeral, and they won't change over time. Where all the config should happen when actually running it (or if the image is already built with the config options you want).